src/HttpCache/EventListener/AddTagsListener.php

```diff
@@ -64,6 +70,33 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
+        $resources = $this->removeDisabledResourcesFromCacheTags($resources);
+
         $response->headers->set('Cache-Tags', implode(',', $resources));
     }
+
+    private function removeDisabledResourcesFromCacheTags($resources)
+    {
+        $resourceCacheHeadersPerResourceClass = [];
+        foreach ($resources as $resource) {
+            if (3 !== \count(explode('/', $resource))) { // simple check if it's an item or collection resource
```

Comment:
just a quick comment, I'll review later tonight or tomorrow but quick question, what is this check doing ?
In your tests you use `/foo` and `/foo/1` so it will never be used.
Moreover as any prefix can be used for resources like `/my/prefix/foo/1` I don't see how it helps.

Reply:
This was supposed to be a check for item operations. I didn't take into account that resource iris can be longer. The problem is that if you give a collection iri to the IriConverter->getItemByIri() it will (rightfully) throw an exception. I need a way to also filter things like foo/1 and ID guessing isn't the way to go. Well this was working in my specific use case but you're right, it's not specific enough. My concrete use case is that I retrieve a collection of resources which have subresources which again have subresources. I reached the max header size pretty quick with it since every resource and subresource had a cache-tag.

Reply:
so filtering  resources just after line 62 where only items are present I think, and collection/subresources in the if line 64 should do the trick no ?